yarl/__init__.py

```diff
@@ -446,7 +447,10 @@ def host(self):
             # fe80::2%Проверка
             # presence of '%' sign means only IPv6 address, so idna is useless.
             return raw
+        return self._get_host_decoded(raw)
 
+    @lru_cache()
+    def _get_host_decoded(self, raw):
```

Comment:
`lru_cache` uses `(self, raw)` as a cache key.
It means that the cache is unique for every URL instance because `self` is different.
Please make `_get_host_decoded` not a URL method but a first-class function in the global namespace to make caching really work.